### PR TITLE
Adding first semantics perf test

### PR DESCRIPTION
**Pull request by goderbauer**
Measures the duration of building the initial semantics tree for the complex_layout app.

### CODE REVIEW

File: dev/benchmarks/complex_layout/test_driver/semantics_perf_test.dart [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    });
+
+    tearDownAll(() async {
+      if (driver != null)
+        driver.close();
+    });
+
+    test('inital tree creation', () async {
+      // Let app become fully idle.
+      await new Future<Null>.delayed(new Duration(seconds: 1));
```
Comment (yjbanov) [inline dev/benchmarks/complex_layout/test_driver/semantics_perf_test.dart]: I think calling `driver.waitFor` would be more robust than waiting for a magical amount of time, as `waitFor` ensures there's nothing going on in the app.

File: dev/devicelab/bin/tasks/complex_layout_semantics_perf.dart [hunk 2]
```diff
@@ -0,0 +1,34 @@
+// Copyright (c) 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter_devicelab/framework/adb.dart';
+import 'package:flutter_devicelab/framework/framework.dart';
+import 'package:flutter_devicelab/framework/utils.dart';
+import 'package:path/path.dart' as p;
+
+void main() {
```
Comment (yjbanov) [inline dev/devicelab/bin/tasks/complex_layout_semantics_perf.dart]: You might need to set the device OS [like so](https://github.com/flutter/flutter/blob/45a4f639c9efb31255a0fdf00de87fa25c857616/dev/devicelab/bin/tasks/complex_layout__start_up.dart#L12)

File: dev/devicelab/lib/framework/utils.dart [hunk 3]
```diff
@@ -208,8 +208,9 @@ Future<int> exec(
   List<String> arguments, {
   Map<String, String> environment,
   bool canFail: false,
+  String workingDirectory,
 }) async {
-  final Process process = await startProcess(executable, arguments, environment: environment);
+  final Process process = await startProcess(executable, arguments, environment: environment, workingDirectory: workingDirectory);
```
Comment (yjbanov) [inline dev/devicelab/lib/framework/utils.dart]: Consider using `inDirectory` instead.

File: packages/flutter_driver/lib/src/extension.dart [hunk 4]
```diff
@@ -271,4 +274,27 @@ class FlutterDriverExtension {
...
+        SchedulerBinding.instance.addPostFrameCallback((Duration d) {
+          completer.complete();
+        });
+        await completer.future;
+      }
+    } else if (!setSemanticsCommand.enabled && _semantics != null) {
+      _semantics.dispose();
+      _semantics = null;
+    }
+    final bool enabled = RendererBinding.instance.pipelineOwner.semanticsOwner != null;
```
Comment (yjbanov) [inline packages/flutter_driver/lib/src/extension.dart]: nit: since the other variable is called `wasEnabled`, perhaps this one should be called `isEnabled`. But also, you could make the expression into a getter, then you wouldn't need a second variable:

```dart
bool get isEnabled => RendererBinding.instance.pipelineOwner.semanticsOwner != null;
bool wasEnabled = isEnabled;
...
return new SetSemanticsResult(wasEnabled != isEnabled);
```

File: packages/flutter_driver/lib/src/message.dart [hunk 5]
```diff
@@ -20,6 +22,7 @@ abstract class Command {
   String get kind;
 
   /// Serializes this command to parameter name/value pairs.
+  @mustCallSuper
```
Comment (yjbanov) [inline packages/flutter_driver/lib/src/message.dart]: \o/

File: packages/flutter_driver/lib/src/semantics.dart [hunk 6]
```diff
@@ -0,0 +1,43 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'message.dart';
+
+/// Enables or disables Semantics.
```
Comment (yjbanov) [inline packages/flutter_driver/lib/src/semantics.dart]: nit: when I see capitalized words in plain English text I immediately think it's either a proper noun or a reference to a type. I'm guessing this is not a proper noun, but if it's a type, consider backquoting it: `Semantics` vs Semantics.

**Review by yjbanov — APPROVED** [review]
LGTM.

File: dev/benchmarks/complex_layout/test_driver/semantics_perf_test.dart (hunk 1 above)
Comment (goderbauer) [inline dev/benchmarks/complex_layout/test_driver/semantics_perf_test.dart]: I can't find any element to wait for, though. Shortly after app app loads the dimensions of some elements change, but I don't see a way to robustly wait for that...

File: dev/devicelab/lib/framework/utils.dart (hunk 3 above)
Comment (goderbauer) [inline dev/devicelab/lib/framework/utils.dart]: Nice!
